guides/get-started-microservices-on-kubernetes/simple-microservice-llm-part1.adoc

```diff
@@ -0,0 +1,274 @@
+                </goals>
+            </execution>
+        </executions>
+    </plugin>
+----
+
+NOTE: The `wildfly-maven-plugin` configuration can be simplified by using link:https://docs.wildfly.org/wildfly-glow/[wildfly-glow]; link:https://docs.wildfly.org/wildfly-glow/[wildfly-glow] inspects your deployment and figures out what `feature-pack` and `layers` to use automatically! just replace the whole `configuration` section with the following:
+[source,xml,subs="normal"]
+----
+<configuration>
+    <discoverProvisioningInfo>
+        <context>cloud</context>
+        <spaces>
+            <space>incubating</space>
+        </spaces>
+    </discoverProvisioningInfo>
+</configuration>
+----
+
+=== Java Classes
+
+Replace the content of the `org.wildfly.examples.GettingStartedEndpoint` class with the following:
+
+.org.wildfly.examples.GettingStartedEndpoint :
+[source,java]
+----
+package org.wildfly.examples;
+
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.message.UserMessage;
+import dev.langchain4j.memory.ChatMemory;
+import dev.langchain4j.memory.chat.MessageWindowChatMemory;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.inject.Inject;
+import jakarta.inject.Named;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+
+@Path("/")
+@RequestScoped
+public class GettingStartedEndpoint {
+    @Inject
+    @Named(value = "ollama")
+    ChatLanguageModel model;
+
+    @GET
+    @Path("/{name}")
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response sayHello(final @PathParam("name") String name) {
+        ChatMemory memory = MessageWindowChatMemory.withMaxMessages(5);
+        UserMessage message1 = UserMessage.from("Hi! my name is " + name);
+        memory.add(message1);
+        AiMessage response1 = model.chat(memory.messages()).aiMessage();
+        memory.add(response1);
+        return Response.ok(response1).build();
+    }
+}
+----
+
+Delete class `org.wildfly.examples.GettingStartedService` which isn't used anymore at this point, since the LLM is now responsible for greeting us!
+
+=== Build the application
+
+[source,bash]
+----
+$ mvn clean package
+...
+[INFO] Copy deployment /home/tborgato/projects/guides/get-started-microservices-on-kubernetes/simple-microservice-llm/target/ROOT.war to /home/tborgato/projects/guides/get-started-microservices-on-kubernetes/simple-microservice-llm/target/server/standalone/deployments/ROOT.war
+[INFO] ------------------------------------------------------------------------
+[INFO] BUILD SUCCESS
+[INFO] ------------------------------------------------------------------------
+[INFO] Total time:  6.694 s
+[INFO] Finished at: 2025-03-19T14:39:09+01:00
+[INFO] ------------------------------------------------------------------------
+----
+
+== Docker Image
```

Comment:
@tommaso-borgato , the wildfly maven plugin has a goal to automatically build an image. I feel that it would greatly simplify the doc.

Reply:
Using a `Dockerfile` is a choice for all guides.